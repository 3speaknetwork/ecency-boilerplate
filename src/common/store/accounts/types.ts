import { ExtendedAccount } from "@esteemapp/dhive";

export interface Account extends ExtendedAccount {}

export interface State {
  data: Account;
  loading: boolean;
  error: boolean;
}

export enum ActionTypes {
  FETCH = "@account/FETCH",
  FETCHED = "@account/FETCHED",
  FETCH_ERROR = "@account/FETCH_ERROR"
}

export interface FetchAction {
  type: ActionTypes.FETCH;
}

export interface FetchedAction {
  type: ActionTypes.FETCHED;
  data: Account;
}

export interface FetchErrorAction {
  type: ActionTypes.FETCH_ERROR;
}

export type Actions = FetchAction | FetchedAction | FetchErrorAction;