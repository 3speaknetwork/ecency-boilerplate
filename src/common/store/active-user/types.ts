import { Account } from "../accounts/types";

export interface ActiveUser {
  name: string;
  data: Account;
}

export enum ActionTypes {
  RELOAD = "@active-user/RELOAD",
  UPDATE = "@active-user/UPDATE",
}

export interface ReloadAction {
  type: ActionTypes.RELOAD;
}

export interface UpdateAction {
  type: ActionTypes.UPDATE;
  data: Account;
}

export type Actions = ReloadAction | UpdateAction;
