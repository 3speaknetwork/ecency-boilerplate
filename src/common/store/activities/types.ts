interface BaseActivity {
    event: "notify";
    source: string;
    target: string;
    timestamp: string;
    extra: any;
}

export interface VoteActivity extends BaseActivity {
    type: "vote",
    extra: {
        permlink: string;
        weight: number;
        title: string | null;
        img_url: string | null;
    }
}

export interface MentionActivity extends BaseActivity {
    type: "mention",
    extra: {
        permlink: string;
        is_post: 0 | 1;
        title: string | null;
        img_url: string | null;
    }
}

export interface FollowActivity extends BaseActivity {
    type: "follow",
    extra: {
        what: string[]
    }
}

export interface ReplyActivity extends BaseActivity {
    type: "reply",
    extra: {
        title: string;
        body: string;
        json_metadata: string;
        permlink: string;
        parent_author: string;
        parent_permlink: string;
        parent_title: string | null;
        parent_img_url: string | null;
    }
}

export interface ReblogActivity extends BaseActivity {
    type: "reblog";
    extra: {
        permlink: string;
        title: string | null;
        img_url: string | null;
    }

}

export interface TransferActivity extends BaseActivity {
    type: "transfer";
    extra: {
        amount: string;
        memo: string;
    }
}


export type Activity = VoteActivity | MentionActivity | FollowActivity | ReplyActivity | ReblogActivity | TransferActivity;
