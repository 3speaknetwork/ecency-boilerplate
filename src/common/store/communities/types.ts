export interface Community {
    about: string,
    avatar_url: string,
    created_at: string,
    description: string,
    flag_text: string,
    id: number,
    is_nsfw: boolean,
    lang: string,
    name: string,
    num_authors: number,
    num_pending: number,
    subscribers: number,
    sum_pending: number,
    team: Array<Array<string>>,
    title: string,
    type_id: number
}

export interface State {
    list: Record<string, Community>,
}

export enum ActionTypes {
    FETCHED = '@communities/FETCHED',
}


export interface FetchedAction {
    type: ActionTypes.FETCHED;
    name: string,
    data: Community
}

export type Actions = FetchedAction; // | .. | ..
