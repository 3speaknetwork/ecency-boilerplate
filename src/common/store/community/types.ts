export interface Community {
  about: string;
  avatar_url: string;
  created_at: string;
  description: string;
  flag_text: string;
  id: number;
  is_nsfw: boolean;
  lang: string;
  name: string;
  num_authors: number;
  num_pending: number;
  subscribers: number;
  sum_pending: number;
  settings?: any;
  team: Array<Array<string>>;
  title: string;
  type_id: number;
}

export interface State extends Community {}

export enum ActionTypes {
  FETCHED = "@communities/FETCHED",
  RESET = "@communities/RESET",
}

export interface FetchedAction {
  type: ActionTypes.FETCHED;
  data: Community;
}

export interface ResetAction {
  type: ActionTypes.RESET;
}

export type Actions = FetchedAction | ResetAction;
