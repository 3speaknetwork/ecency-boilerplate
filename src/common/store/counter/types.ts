export interface State {
    val: number
}

export enum ActionTypes {
    SET_COUNTER = '@counter/SET_COUNTER',
    INCREMENT_COUNTER = '@counter/INCREMENT_COUNTER',
    DECREMENT_COUNTER = '@counter/DECREMENT_COUNTER',
}

export interface SetAction {
    type: ActionTypes.SET_COUNTER;
    value: string;
}

export interface IncrementAction {
    type: ActionTypes.INCREMENT_COUNTER;
}

export interface DecrementAction {
    type: ActionTypes.DECREMENT_COUNTER;
}

export type Actions = SetAction | IncrementAction | DecrementAction
