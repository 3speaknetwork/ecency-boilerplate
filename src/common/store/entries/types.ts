import { LocationChangeAction } from "../common";

export interface EntryBeneficiaryRoute {
  account: string;
  weight: number;
}

export interface EntryVote {
  voter: string;
  rshares: string;
}

export interface EntryStat {
  flag_weight: number;
  gray: boolean;
  hide: boolean;
  total_votes: number;
}

export interface JsonMetadata {
  tags: string[];
  app: string;
  canonical_url?: string;
}

export interface Entry {
  active_votes: EntryVote[];
  author: string;
  author_payout_value: string;
  author_reputation: number;
  author_role?: string;
  author_title?: string;
  beneficiaries: EntryBeneficiaryRoute[];
  blacklists: string[];
  body: string;
  category: string;
  children: number;
  community?: string;
  community_title?: string;
  created: string;
  curator_payout_value: string;
  depth: number;
  is_paidout: boolean;
  json_metadata: JsonMetadata;
  max_accepted_payout: string;
  net_rshares: number;
  parent_author?: string;
  parent_permlink?: string;
  payout: number;
  payout_at: string;
  pending_payout_value: string;
  percent_steem_dollars: number;
  permlink: string;
  post_id: number;
  promoted: string;
  reblogged_by?: string[];
  replies: any[];
  stats: EntryStat;
  title: string;
  updated: string;
  url: string;
}

export interface EntryGroup {
  entries: Entry[];
  error: string | null;
  loading: boolean;
  hasMore: boolean;
}

export interface State extends Record<string, EntryGroup> {}

// TODO: Implement UPDATE action

export enum ActionTypes {
  FETCH = "@entries/FETCH",
  FETCHED = "@entries/FETCHED",
  FETCH_ERROR = "@entries/FETCH_ERROR",
  INVALIDATE = "@entries/INVALIDATE",
}

export interface FetchAction {
  type: ActionTypes.FETCH;
  groupKey: string;
}

export interface FetchErrorAction {
  type: ActionTypes.FETCH_ERROR;
  groupKey: string;
  error: string;
}

export interface FetchedAction {
  type: ActionTypes.FETCHED;
  groupKey: string;
  entries: Entry[];
  hasMore: boolean;
}

export interface InvalidateAction {
  type: ActionTypes.INVALIDATE;
  groupKey: string;
}

export type Actions = LocationChangeAction | FetchAction | FetchedAction | FetchErrorAction | InvalidateAction;
