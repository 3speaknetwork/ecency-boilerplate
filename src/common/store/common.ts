export enum CommonActionTypes {
  INIT = "@@INIT",
  LOCATION_CHANGE = "@@router/LOCATION_CHANGE",
}

export interface LocationChangeAction {
  type: CommonActionTypes.LOCATION_CHANGE;
  payload: {
    location: {
      pathname: string;
    };
  };
}

export interface InitAction {
  type: CommonActionTypes.INIT;
}

export const locationChangeAct = (pathname: string): LocationChangeAction => {
  return {
    type: CommonActionTypes.LOCATION_CHANGE,
    payload: {
      location: {
        pathname,
      },
    },
  };
};
