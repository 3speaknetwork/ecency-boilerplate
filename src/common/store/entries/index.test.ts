import reducer, {
    initialState,
    fetchAct,
    fetchErrorAct,
    fetchedAct,
    invalidateAct
} from './index';

import {locationChangeAct} from '../common';

import {State} from './types';


let state: State = initialState;

it('1- default state', () => {
    expect(state).toMatchSnapshot();
});

it('2- location change. should create a group with default filter.', () => {
    state = reducer(state, locationChangeAct('/'));
    expect(state).toMatchSnapshot();
});

it('3- location change. should create a group with the filter and tag given.', () => {
    state = reducer(state, locationChangeAct('/trending/bitcoin'));
    expect(state).toMatchSnapshot();
});

it('4- location change with invalid filter. state should state same.', () => {
    state = reducer(state, locationChangeAct('/trendinx/bitcoin'));
    expect(state).toMatchSnapshot();
});

it('5- start fetching. should toggle "loading"', () => {
    const act = fetchAct('hot');
    state = reducer(state, act);
    expect(state).toMatchSnapshot();
});

it('6- fetch error. should toggle "loading" and add error object to the group.', () => {
    const act = fetchErrorAct('hot', 'fetch error');
    state = reducer(state, act);
    expect(state).toMatchSnapshot();
});

it('7- start fetching.', () => {
    const act = fetchAct('trending-bitcoin');
    state = reducer(state, act);
    expect(state).toMatchSnapshot();
});


it('8- fetched.', () => {
    const entries = [
        {
            author: 'lorem',
            permlink: 'lorem-ipsum-dolor-sit-amet'
        },
        {
            author: 'consectetur',
            permlink: 'consectetur-adipiscing elit-in-vel-enim-libero'
        },
        {
            author: 'nam',
            permlink: 'tempor-nibh-eget-enim-vulputate'
        },
        {
            author: 'vivamus',
            permlink: 'vivamus-ornare-at-tortor-eu-vestibulum'
        }
    ];

    // @ts-ignore
    const act = fetchedAct('trending-bitcoin', entries, true);
    state = reducer(state, act);
    expect(state).toMatchSnapshot();
});

it('9- start fetching again (simulates user scrolling down)', () => {
    const act = fetchAct('trending-bitcoin');
    state = reducer(state, act);
    expect(state).toMatchSnapshot();
});

it('10- fetched. should ignore posts already in store.', () => {
    const entries = [
        {
            author: 'vivamus',
            permlink: 'vivamus-ornare-at-tortor-eu-vestibulum'
        },
        {
            author: 'donec',
            permlink: 'donec-bibendum-tortor-ultricies'
        }
    ];

    // @ts-ignore
    const act = fetchedAct('trending-bitcoin', entries, false);
    state = reducer(state, act);
    expect(state).toMatchSnapshot();
});

it('11- invalidate. should reset the group.', () => {
    const act = invalidateAct('trending-bitcoin');
    state = reducer(state, act);
    expect(state).toMatchSnapshot();
});
