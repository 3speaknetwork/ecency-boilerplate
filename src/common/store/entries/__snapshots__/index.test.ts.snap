// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1- default state 1`] = `Object {}`;

exports[`2- location change. should create a group with default filter. 1`] = `
Object {
  "hot": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
  },
}
`;

exports[`3- location change. should create a group with the filter and tag given. 1`] = `
Object {
  "hot": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
  },
  "trending-bitcoin": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
  },
}
`;

exports[`4- location change with invalid filter. state should state same. 1`] = `
Object {
  "hot": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
  },
  "trending-bitcoin": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
  },
}
`;

exports[`5- start fetching. should toggle "loading" 1`] = `
Object {
  "hot": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": true,
  },
  "trending-bitcoin": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
  },
}
`;

exports[`6- fetch error. should toggle "loading" and add error object to the group. 1`] = `
Object {
  "hot": Object {
    "entries": Array [],
    "error": [Error: fetch error],
    "hasMore": false,
    "loading": false,
  },
  "trending-bitcoin": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
  },
}
`;

exports[`7- start fetching. 1`] = `
Object {
  "hot": Object {
    "entries": Array [],
    "error": [Error: fetch error],
    "hasMore": false,
    "loading": false,
  },
  "trending-bitcoin": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": true,
  },
}
`;

exports[`8- fetched. 1`] = `
Object {
  "hot": Object {
    "entries": Array [],
    "error": [Error: fetch error],
    "hasMore": false,
    "loading": false,
  },
  "trending-bitcoin": Object {
    "entries": Array [
      Object {
        "author": "lorem",
        "permlink": "lorem-ipsum-dolor-sit-amet",
      },
      Object {
        "author": "consectetur",
        "permlink": "consectetur-adipiscing elit-in-vel-enim-libero",
      },
      Object {
        "author": "nam",
        "permlink": "tempor-nibh-eget-enim-vulputate",
      },
      Object {
        "author": "vivamus",
        "permlink": "vivamus-ornare-at-tortor-eu-vestibulum",
      },
    ],
    "error": null,
    "hasMore": true,
    "loading": false,
  },
}
`;

exports[`9- start fetching again (simulates user scrolling down) 1`] = `
Object {
  "hot": Object {
    "entries": Array [],
    "error": [Error: fetch error],
    "hasMore": false,
    "loading": false,
  },
  "trending-bitcoin": Object {
    "entries": Array [
      Object {
        "author": "lorem",
        "permlink": "lorem-ipsum-dolor-sit-amet",
      },
      Object {
        "author": "consectetur",
        "permlink": "consectetur-adipiscing elit-in-vel-enim-libero",
      },
      Object {
        "author": "nam",
        "permlink": "tempor-nibh-eget-enim-vulputate",
      },
      Object {
        "author": "vivamus",
        "permlink": "vivamus-ornare-at-tortor-eu-vestibulum",
      },
    ],
    "error": null,
    "hasMore": true,
    "loading": true,
  },
}
`;

exports[`10- fetched. should ignore posts already in store. 1`] = `
Object {
  "hot": Object {
    "entries": Array [],
    "error": [Error: fetch error],
    "hasMore": false,
    "loading": false,
  },
  "trending-bitcoin": Object {
    "entries": Array [
      Object {
        "author": "lorem",
        "permlink": "lorem-ipsum-dolor-sit-amet",
      },
      Object {
        "author": "consectetur",
        "permlink": "consectetur-adipiscing elit-in-vel-enim-libero",
      },
      Object {
        "author": "nam",
        "permlink": "tempor-nibh-eget-enim-vulputate",
      },
      Object {
        "author": "vivamus",
        "permlink": "vivamus-ornare-at-tortor-eu-vestibulum",
      },
      Object {
        "author": "donec",
        "permlink": "donec-bibendum-tortor-ultricies",
      },
    ],
    "error": null,
    "hasMore": false,
    "loading": false,
  },
}
`;

exports[`11- invalidate. should reset the group. 1`] = `
Object {
  "hot": Object {
    "entries": Array [],
    "error": [Error: fetch error],
    "hasMore": false,
    "loading": false,
  },
  "trending-bitcoin": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
  },
}
`;
